litestar/response/base.py

```diff
@@ -24,28 +22,172 @@
         ResponseHeaders,
         Scope,
         Send,
+        Serializer,
         TypeEncodersMap,
     )
 
+__all__ = ("ASGIResponse", "Response")
+
 T = TypeVar("T")
 
 
+class ASGIResponse:
+    """A low-level ASGI response class."""
+
+    __slots__ = (
+        "background",
+        "body",
+        "content_length",
+        "encoded_headers",
+        "encoding",
+        "is_head_response",
+        "status_code",
+    )
+
+    def __init__(
+        self,
+        *,
+        background: BackgroundTask | BackgroundTasks | None = None,
+        body: bytes = b"",
+        content_length: int | None = None,
+        cookies: list[Cookie] | None = None,
+        encoded_headers: list[tuple[bytes, bytes]] | None = None,
+        encoding: str = "utf-8",
+        headers: dict[str, Any] | None = None,
```

Comment:
I don't think this should have both header types. A single parameter that accepts encoded headers should be what we want. This is intended to be a low level implementation of the ASGI response protocol, so encoding and such should happen outside.

Reply:
I do agree, and I did start with that but landed here for practical reasons.

First, centralizing the encoding logic - although that can be done in other ways so don't worry about that.

Second, in ASGIFileResponse, a few headers are set right before `http.response.start` is sent due to the source of those values potentially coming from an awaitable that is resolved within the `start_response()` coroutine. Given that those are unconditionally set by `ASGIFileResponse`, I `pop()` them from `headers` in `__init__()`.